cbmc/proofs/IotMqtt_Connect/README.md

```diff
@@ -0,0 +1,141 @@
+
+* libraries/standard/common/include/iot_linear_containers.h
+  IotListDouble_FindFirstMatch	
+  * Always called with a nonnull isMatch parameter
+
+* libraries/standard/mqtt/src/iot_mqtt_subscription.c
+  \_topicMatch
+  * _IotMqtt_AddSubscriptions sets pParam->exactMatchOnly == true, so _topicFilterMatch is never called
+
+* libraries/standard/mqtt/src/iot_mqtt_helper.c
+  \_encodeUserNameAndMetrics
+  * At this point pConnectInfo->userNameLength is smaller than (UINT16_MAX - AWS_IOT_METRICS_USERNAME_LENGTH)
+
+* libraries/standard/mqtt/src/iot_mqtt_validate.c
+  \_validatePublish
+  * Always called with a nonnull pPublishInfo parameter
+
+* libraries/standard/mqtt/src/iot_mqtt_operation.c
+  \_IotMqtt_DestroyOperation
+  * Notify calls DestroyOperation with a linked operation.
+
+* libraries/standard/mqtt/src/iot_mqtt_api.c
+  \_IotMqtt_IncrementConnectionReferences
+  * pMqttConnection->disconnected is false
+
+* libraries/standard/mqtt/src/iot_mqtt_helper.c
+  \_encodeRemainingLength
+  * remainingLength is smaller than 128 (higher sizes lead to way higher verification time)
+
+* libraries/standard/mqtt/src/iot_mqtt_operation.c
+  \_IotMqtt_DecrementOperationReferences
+  * IotTaskPool_TryCancel always return bad parameter, because we do not create a TaskPool job.
+
+* libraries/standard/mqtt/src/iot_mqtt_operation.c
+  \_initializeOperation
+  * The operation is always waitable
+
+* libraries/standard/mqtt/src/iot_mqtt_operation.c
+  \_IotMqtt_CreateOperation
+  * pMqttConnection->disconnected is true; therefore, _IotMqtt_IncrementConnectionReferences always return false
+
+* libraries/standard/mqtt/src/iot_mqtt_api.c
+  \_createKeepAliveOperation
+  * _getMqttPingreqSerializer always return IOT_MQTT_SUCCESS
+
+* libraries/standard/mqtt/src/iot_mqtt_operation.c
+  \_completePendingSend
+  * pOperation->u.operation.periodic.retry.limit always 0
+
+* libraries/standard/mqtt/src/iot_mqtt_operation.c
+  \_IotMqtt_ProcessSend
+  * pOperation->u.operation.periodic.retry.limit always 0
+  * pOperation->u.operation.type != IOT_MQTT_DISCONNECT
+  * operation is always waitable
+  * _completePendingSend always return destroyOperation == false
+
+* libraries/standard/mqtt/src/iot_mqtt_api.c
+  \IotMqtt_Wait
+  * operation is always initialized at this point
+  * all parameters are valid
+  * pMqttConnection->disconnected == false
+
+* libraries/standard/mqtt/src/iot_mqtt_api.c
+  \_IotMqtt_DecrementConnectionReferences
+  * pMqttConnection->references always != 0, so the connection won't be destroyed
+  * destroyConnection == false, because pMqttConnection->references always != 0
+
+* libraries/standard/mqtt/src/iot_mqtt_api.c
+  \_waitForOperation
+
+  * operation->u.operation.type != IOT_MQTT_SUBSCRIBE
+
+* libraries/standard/mqtt/src/iot_mqtt_api.c
+  IotMqtt_strerror
+
+  * Never returns the following messages: IOT_MQTT_SUCCESS, IOT_MQTT_INIT_FAILED, IOT_MQTT_BAD_RESPONSE, IOT_MQTT_SERVER_REFUSED, IOT_MQTT_RETRY_NO_RESPONSE, and default
```

Comment:
Format lines to 80 characters.  This is a readme file, should be easy to read in a text editor and not just formatted markdown.

Reply:
Done.